### PR TITLE
Pas 2 create order

### PR DESCRIPTION
**Pull request by jakzda**
branch name change

### CODE REVIEW

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 1]
```diff
@@ -1,51 +1,127 @@
...
+
 export const CreateOrder: FC = () => {
   const { register, handleSubmit } = useForm();
   const { createOrder } = useOrdersFacade();
 
+  const [restaurant, setRestaurant] = useState<string>('');
+  const [menu, setMenu] = useState<string>('');
+  const [isValid, setIsValid] = useState<boolean>(false);
+  const [isValidInput, setIsValidInput] = useState<boolean>(false);
+  const [DeliveryPrice, setDeliveryPrice] = useState<string>('');
```
Comment (Isterius) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:18]: `deliveryPrice`

Comment (Isterius) [reply apps/frontend/src/app/CreateOrder/CreateOrder.tsx:18]: and again - why do we need this?

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 2]
```diff
@@ -1,51 +1,127 @@
...
   const onSubmit = (data: CreateOrderDto) => {
-    createOrder({ ...data, orderAt: new Date().toISOString() })
-      .then((params: OrderModel) => history.push(`/`))
+    const newPrice = data.shippingCents
+      ? +formatter.format(data.shippingCents)
+      : -1;
+    createOrder({
+      ...data,
+      orderAt: new Date().toISOString(),
+      shippingCents: newPrice * 100,
```
Comment (Isterius) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:32]: what if `shippingCents` is not provided? we're passing `-100` here? 🤔

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 3]
```diff
@@ -1,51 +1,127 @@
...
+              fullWidth: true,
+            }}
+            {...register('from')}
+            onChange={handleChangeRestaurant}
+            type="text"
+            required
+          />
+          <TextField
+            margin="normal"
+            error={isValidInput}
```
Comment (Isterius) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:84]: this basically means that `error === true` when input is valid `isValidInput === true` 
something is wrong with names

**Review by Isterius — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx (hunk 2 above)
Comment (jakzda) [reply apps/frontend/src/app/CreateOrder/CreateOrder.tsx:32]: I had an idea to add condition in other components    shippingCents>=0 shippingCents : 'shipping price is not added'. This way turned off errors with 'object might be undefined or null'

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 4]
```diff
@@ -1,51 +1,124 @@
...
 import type { CreateOrderDto, OrderModel } from '@pasnik/api/data-transfer';
 import { useOrdersFacade } from '@pasnik/orders-data-access';
+import { Alert, Button, InputAdornment, Paper, TextField } from '@mui/material';
+import { formatter, validUrl } from '../../utils';
+
 export const CreateOrder: FC = () => {
   const { register, handleSubmit } = useForm();
   const { createOrder } = useOrdersFacade();
 
+  const [restaurant, setRestaurant] = useState<string>('');
```
Comment (lukszymanski) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:14]: hmm react-hook-form should automatic handle the onChange and form state so you dont need to have to create controller for each input

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 5]
```diff
@@ -1,51 +1,124 @@
...
+  const [menu, setMenu] = useState<string>('');
+  const [formError, setFormError] = useState<boolean>(false);
+  const [isValidInput, setIsValidInput] = useState<boolean>(false);
+  const [deliveryPrice, setDeliveryPrice] = useState<string>('');
+
+  const deliveryRef = useRef<HTMLInputElement>(null);
```
Comment (lukszymanski) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:20]: why this variable is used?

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 6]
```diff
@@ -1,51 +1,124 @@
...
+
   const [error, setError] = useState<string | null>(null);
```
Comment (lukszymanski) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:22]: error state can be also taken from react-hook-forms, yup.js schema validation can be really useful here :)

**Review by Isterius — APPROVED** [review]

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx (hunk 6 above)
Comment (jakzda) [reply apps/frontend/src/app/CreateOrder/CreateOrder.tsx:53]: Okey, I made error handling by react-hook-forms but I left error state still in component for axios error handling.

File: apps/frontend/src/app/CreateOrder/CreateOrder.tsx [hunk 7]
```diff
@@ -1,51 +1,127 @@
...
+        }
+        return false;
+      }),
+    shippingCents: yup
+      .string()
+      .test(
+        'maxDigitsAfterDecimal',
+        'number field must have 2 digits after decimal or less',
+        (number: string | undefined) => {
+          console.log(number);
```
Comment (Isterius) [inline apps/frontend/src/app/CreateOrder/CreateOrder.tsx:33]: console.log

**Review by Isterius — CHANGES_REQUESTED** [review]

**Review by Isterius — APPROVED** [review]

**Pull request merged**